tests/backup/backup_license_test.go

```diff
@@ -0,0 +1,201 @@
+			destClusterStatus, err = Inst().Backup.GetClusterStatus(orgID, destinationClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", destinationClusterName))
+			dash.VerifyFatal(destClusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", destinationClusterName))
+		})
+		Step("Getting the total number of worker nodes in source and destination cluster", func() {
+			log.InfoD("Getting the total number of worker nodes in source and destination cluster")
+			sourceClusterWorkerNodes = node.GetWorkerNodes()
+			log.InfoD("Total number of worker nodes in source cluster are %v", len(sourceClusterWorkerNodes))
+			totalNumberOfWorkerNodes = append(totalNumberOfWorkerNodes, sourceClusterWorkerNodes...)
+			log.InfoD("Switching cluster context to destination cluster")
+			SetDestinationKubeConfig()
+			destinationClusterWorkerNodes = node.GetWorkerNodes()
+			log.InfoD("Total number of worker nodes in destination cluster are %v", len(destinationClusterWorkerNodes))
+			totalNumberOfWorkerNodes = append(totalNumberOfWorkerNodes, destinationClusterWorkerNodes...)
+			log.InfoD("Total number of worker nodes in source and destination cluster are %v", len(totalNumberOfWorkerNodes))
+			log.InfoD("Switching cluster context back to source cluster")
+			err := SetSourceKubeConfig()
+			log.FailOnError(err, "Switching context to source cluster")
+		})
+		Step("Verifying the license count after adding source and destination clusters", func() {
+			log.InfoD("Verifying the license count after adding source and destination clusters")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			licenseInspectResponse, err = Inst().Backup.InspectLicense(ctx, licenseInspectRequestObject)
+			log.FailOnError(err, "Getting the license for organisation %v", orgID)
+			licenseResponseInfoFeatureInfo = licenseInspectResponse.GetLicenseRespInfo().GetFeatureInfo()
+			dash.VerifyFatal(licenseResponseInfoFeatureInfo[0].Consumed, int64(len(totalNumberOfWorkerNodes)), fmt.Sprintf("Verifying the license count when source cluster with %d worker nodes and destination cluster with %d worker nodes are added to backup", len(sourceClusterWorkerNodes), len(destinationClusterWorkerNodes)))
+		})
+		Step("Applying label portworx.io/nobackup=true to worker nodes and verifying the license count", func() {
+			log.InfoD("Applying label portworx.io/nobackup=true to one of the worker node on source cluster and verifying the license count")
+			err := Inst().S.AddLabelOnNode(sourceClusterWorkerNodes[0], "portworx.io/nobackup", "true")
+			if err == nil {
+				sourceClusterWorkerNodeLabelApplied = true
+				sourceClusterWorkerNodeLabelRemoved = false
+			}
+			log.FailOnError(err, fmt.Sprintf("Failed to apply label portworx.io/nobackup=true to worker node %v", sourceClusterWorkerNodes[0].Name))
```

Comment:
You can move the body of the if condition to _AFTER_ the `log.FailOnError`, right? Then you won't need to check for that condition as, if `err` was `nil`, then `log.FailOnError` won't fail and will execute the rest of the code (which is the same behaviour as what you're doing). Like:
```go
log.FailOnError(err, fmt.Sprintf("Failed to apply label portworx.io/nobackup=true to worker node %v", sourceClusterWorkerNodes[0].Name))
sourceClusterWorkerNodeLabelApplied = true
sourceClusterWorkerNodeLabelRemoved = false
```

**You can use the same method in other places.**

Reply:
corrected